hvpy/tests/test_get_jp2_image.py

```diff
@@ -0,0 +1,19 @@
+from datetime import datetime
+
+from hvpy.api_groups.jpeg2000.get_jp2_image import getJP2ImageInputParameters
+from hvpy.core import execute_api_call
+from hvpy.io import BASE_URL
+
+URL = BASE_URL + "getJP2Image/"
```

Comment:
Let's move this URL into getJP2ImageInputParameters.
Per the design, all necessary fields should be included in the InputParameters class so that the only thing passed to `execute_api_call` is an instance of `HvpyParameters`

Reply:
I don't know if this is a too clever approach, we can do:
```py
class HvpyParameters(BaseModel):
    def dict(self):
        d = super().dict()
        # Pydantic doesn't allow using lower case 'json' as a field,
        # so override it here.
        if "Json" in d:
            d["json"] = d["Json"]
            del d["Json"]
        return d

    def get_output_type(self):
        # For now just return the String output
        return OutputType.String

    def url(self):
        return BASE_URL + self.__class__.__name__[:-15] + "/"
```

We can remove `InputParameters` from `<endpoint>InputParameters` and append it to BASE_URL with "/".

Example:
```py
print(getJP2ImageInputParameters(date=datetime(2022, 1, 1, 23, 59, 59), sourceId=14).url())
# Output:
# https://api.helioviewer.org/v2/getJP2Image/
```

Reply:
It is quite clever... I'm ok with it. It keeps us from having to define the url on every class.
As long as we're consistent with naming as we should be it will work.
Unit tests will catch any one that doesn't work.
👍